sql/sql_class.cc

```diff
@@ -2126,6 +2128,9 @@ void THD::reset_killed()
     killed_err= 0;
     mysql_mutex_unlock(&LOCK_thd_kill);
   }
+#ifdef WITH_WSREP
+  wsrep_killed = false;
+#endif /* WITH_WSREP */
```

Comment:
Apparently, no mutex is protecting `wsrep_killed` here.

Reply:
LOCK_thd_kill mutex would protect from concurrent awake() call, does this really need to be outside of that if and mutex protection ?

Reply:
I did not analyze all code paths to here so not sure if LOCK_thd_data is already locked, I would recommend either

mysql_mutex_assert_not_owner(&victim_thd->LOCK_thd_data);
mysql_mutex_lock(&LOCK_thd_kill);
wsrep_killed= false;
mysql_mutex_unlock(&LOCK_thd_kill);

or
mysql_mutex_assert_owner(&victim_thd->LOCK_thd_data);
wsrep_killed= false;